### ISSUE TITLE
Incompatible license

### ISSUE THREAD

**Issue by leamas** [Hakansv/AIS_VD_pi#2]
The README states that the overall license is GPL version 2 or later a. k. a. GPLv2+. However, here is some files which are GPLv3+. At least in shipdriver there is also files with licenses which are not compatible with GPLv2.

The basic solution to this is to declare the overall license to GPLv3+. It fits all the requirements.

I suggest that you look into shipdriver's README which has a section on copyright and licensing. If it's ok with you I think it can be used also here. In any case, do not change any license marking in a source file which is not copyrighted by you only, it's a legal disaster.

**Comment by leamas** [comment]
hm... If you relicense using GPLv3+ you should also change the file COPYING which now is for GPLv2. The GPLv3 version is available in many places, the shipdriver repo's COPYING is perhaps the closest at hand.

**Comment by Hakansv** [comment]
This is nothing I've been really aware of or by any sense of consciousness looked into. The inconsistency is most probably a result of me copying files from several plugins in order to get the ci and PIM process to work. Of course I agree there would be a consistent way of referring to a license. I have low awareness of the license conditions so I will follow what advises I can get to make it consistent.
The plugin's base author is RooieDirk as of the readme file. I continued the work this year and the aim now is to get it into the plugin manager process. I may first discus with Dirk and see if he will be copywriter for this year as well.
If the README file would refer to GPLv3 as Shipdriver it's no problem for me. If so do I understand you correct if all individual files in that case would keep the license text as it is except the COPYING file?

Many thanks for your attention about this.
Håkan

**Comment by leamas** [comment]
You must not change the license text of any file, this is a legal condition. Only the copyright holder can do that.

Now, the simple truth is that some files are "GPL version 2 or later", and some "GPL version 3 or later". Since "GPL version 3 or later" actually is an example of "GPL version 2 or later" you can publish the complete thing using "GPL version 3 or later" without actually discussing with anyone, all license conditions are fulfilled by this overall license.

The mess becomes more complex when talking about the libraries. Some of them are using licenses like Apache-2 which actually cannot be combined with GPL-2. However, it *can* be combined with GPL-3, so also this possible issue is resolved by using an overall GPL-3 license.

So, in short: Current situation is a legal mess. However, changing the overall license to GPL-3+ resolves this in a simple way. You can do this on your own, there is no need to change any license in the source files for that.

**Comment by Hakansv** [comment]
And "my own" is the README and the COPYING  files then, or?

**Comment by leamas** [comment]
README is your own. The COPYING file is a verbatim copy of the license, currently GPL2 but should be updated to GPL-3. You must not modify it.

**Comment by leamas** [comment]
BTW: don't take me wrong, you are of course free to change file besides COPYING. The exception is just the copyright notice which you cannot tamper with. However, you can add your own copyright together with the existing if/when you make changes.

Yes, free software also has some legal conditions. They are actually occasionally quite complicated.

**Comment by RooieDirk** [comment]
Hakan,
I did the same as you did, copying some files together, and add/change where needed. As far as I have something to say about it, as partly the original author, do as ever you like. I don't mind if you remove my name completely from the source files.

**Comment by leamas** [comment]
@RooieDirk : this is actually  bit more complicated than so. I do think Håkan would be on the safe side if you did a PR where you removed your  copyright and name. But that's a lot of work for no gain. 

Google fo re-licensing, it will show you the horrors involved in changing these things...

The simple and safe route is to do what everyone else do: Keep the copyright, name  and license. And change anything else as you like. It's much simpler than to fiddle with the copyright.

**Comment by Hakansv** [comment]
Dirk..
You have done the most. I've just "tuned" after my mind and "struggled" with the plugin manager part. (Close to the same time?)
The latest Tarballs are here.

If accepted by Alec; what about this?:

In readme.md:
Copyright and licensing
This software is Copyright (c) RooieDirk 2017-2020. Hakansv 2021-2022. It is distributed

In plugin.cake:
set(CPACK_PACKAGE_CONTACT "Hakan Svensson")
...........
set(PKG_AUTHOR "Dirk R and Hakan S")


But.. If you don't want to be a part of what I've put together I can take your name out but I'd would be glad if you're mentioned as well.

Thanks.
Håkan

**Comment by leamas** [comment]
There is no need to mention all of the people involved when you put together things. If you look at shipdriver's README it says: 

> This software is Copyright (c) Mike Rossiter 2017-2020. It is distributed
under the terms of the Gnu Public License version 3 or, at your option,
any later version. See the file COPYING for details.
[...]
> The software contains open-source licensed files under different licenses
including Expat, Apache-2  and Libpng. See the source files for details.

This is basically all there is from a legal point of view. If you want to express kudos so fine, but it's better off in a separate paragraph so you can have the legal stuff simple and clean.

You are the one publishing it, you have the overall copyright and license. It's your doing. Keep it simple, assign a copyright and license it in your name. After all, you are the person who applies the GPL-3 license to everything.

**Comment by leamas** [comment]
Ouch. I  must be tired. Your proposal is actually fine. Please disregard my novel
